src/data.rs

```diff
@@ -5,6 +5,64 @@ use crate::{
 use mongodb::bson::{self, Document};
 use serde::{Deserialize, Serialize};
 use std::{convert::TryFrom, fmt::Display, str::FromStr};
+use strum::{IntoEnumIterator, EnumIter};
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, EnumIter)]
+pub enum BinaryType {
+    ImageJPEG,
+    ImagePNG,
+    ImageGIF,
+    ImageAPNG,
+    ImageAVIF,
+    ImageSVG,
+    ImageWEBP,
+    VideoMP4,
+    VideoMPEG,
+}
+
+impl TryFrom<&str> for BinaryType {
+    type Error = CryptoError;
+
+    fn try_from(s: &str) -> Result<BinaryType, CryptoError> {
+        match s {
+            "image/jpeg" => Ok(BinaryType::ImageJPEG),
+            "image/png" => Ok(BinaryType::ImagePNG),
+            "image/gif" => Ok(BinaryType::ImageGIF),
+            "image/apng" => Ok(BinaryType::ImageAPNG),
+            "image/avif" => Ok(BinaryType::ImageAVIF),
+            "image/svg+xml" => Ok(BinaryType::ImageSVG),
+            "image/webp" => Ok(BinaryType::ImageWEBP),
+            "video/mp4" => Ok(BinaryType::VideoMP4),
+            "video/mpeg" => Ok(BinaryType::VideoMPEG),
+            _ => Err(CryptoError::NotDeserializableToBaseDataType)
+        }
+    }
+}
+
+impl Display for BinaryType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f,
+               "{}",
+               match self {
+                   BinaryType::ImageJPEG => "image/jpeg",
+                   BinaryType::ImagePNG => "image/png",
+                   BinaryType::ImageGIF => "image/gif",
+                   BinaryType::ImageAPNG => "image/apng",
+                   BinaryType::ImageAVIF => "image/avif",
+                   BinaryType::ImageSVG => "image/svg+xml",
+                   BinaryType::ImageWEBP => "image/webp",
+                   BinaryType::VideoMP4 => "video/mp4",
+                   BinaryType::VideoMPEG => "video/mpeg",
+               }
+        )
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct BinaryData {
+    pub binary: String,
```

Comment:
I don't think this is an appropriate data type for the binary content. I believe a string HAS to be UTF-8 bytes, I'm surprised this works at all actually. I think a Vec<u8> would be more appropriate.